components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1063,6 +1064,10 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 }).catch();
             }
 
+            if (User.isOnboardingUser(user) && (!IPrefixContextParserContext.is(context) || !context.specifiedIDE)) {
```

Comment:
why `User.isOnboardingUser(user) && !WithReferrerContext.is(context)` does not work?